app/src/main/java/edu/uc/kovaciad/slicetracker/ui/main/MainFragment.kt

```diff
@@ -9,24 +9,22 @@ import edu.uc.kovaciad.slicetracker.R
 import edu.uc.kovaciad.slicetracker.dto.Printer
 
 
-class MainFragment : Fragment() {
+class MainFragment : SuperFragment() {
 
     companion object {
         fun newInstance() = MainFragment()
     }
 
-    private lateinit var viewModel: MainViewModel
-
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?,
-    ): View {
+    ): View? {
         setHasOptionsMenu(true)
         return inflater.inflate(R.layout.main_fragment, container, false)
     }
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
```

Comment:
Changing from onViewCreated() to onActivityCreated(). This was done to prevent the crashing of the app and facilitate the use of viewModel in fragments after the new super fragments view initialization. Because onActivityCreated() occurs after view initialization, it was necessary to use this convention or else the use of the viewModel would not work correctly as it was not initialized yet in onViewCreated()